waspc/src/Wasp/SemanticVersion.hs

```diff
@@ -0,0 +1,31 @@
+module Wasp.SemanticVersion
+  ( SemanticVersion (..),
+    VersionBounds (..),
+    isVersionInBounds,
+  )
+where
+
+import Text.Printf (printf)
+
+data SemanticVersion = SemanticVersion
```

Comment:
Naming wise -> due to how Haskell does imports/exports, I tend to go with one of two approaches:
1. Give such names, to everything in the module, that it is clear from just the names what we are dealing with. That might mean prefixing all with `SemanticVersion`. In this case, you assume this module might be imported unqualified in the rest of the code.
2. Give shorter names, so it is not completely clear just from them what it is about, the context. In this case, you assume that this module will be imported qualified in the rest of the code.

In this case, I would probably go with the option 2.

So your names would be:
```
Version
VersionBounds
Exact or ExactBounds
CompatibleWith or CompatibleWithBounds
isVersionInBounds
```

And in the rest of the code you would import it qualified as `Semantic` or `SV` or smth like that.

Reply:
This makes sense, I'll change it.